plugins/bundle/config.go

```diff
@@ -185,6 +198,15 @@ func (c *Config) validateAndInjectDefaults(services []string, keys map[string]*k
 		}
 
 		source.Service, err = c.getServiceFromList(source.Service, services)
+
+		if source.Trigger == nil {
+			source.Trigger = trigger
+		} else {
+			if *source.Trigger != *trigger {
+				return fmt.Errorf("invalid configuration for bundle %q: trigger mode mismatch", name)
```

Comment:
[nit] We could include the difference, i.e. "discovery has mode x, bundle has y", for easier debugging if this ever happens for someone?

Reply:
Updated the error message.